src/train.py

```diff
@@ -54,8 +54,7 @@
 shared_dim = opt.dim * 2 ** opt.n_downsample
 
 # Initialize generator and discriminator
-shared_E = ResidualBlock(features=shared_dim)
-encoder = Encoder(dim=opt.dim, in_channels=opt.channels, n_downsample=opt.n_downsample, shared_block=shared_E)
+encoder = Encoder(dim=opt.dim, in_channels=opt.channels, n_downsample=opt.n_downsample, final_block=ResidualBlock(features=shared_dim))
```

Comment:
it would be better if we edit the model itself to call `ResidualBlock` since there is only one encoder in this case.